js/src/vector.ts

```diff
@@ -348,7 +350,15 @@ export class FixedSizeBinaryVector extends FlatVector<FixedSizeBinary> {
     }
 }
 
+const utf8Encoder = new TextEncoder('utf-8');
+
 export class Utf8Vector extends ListVectorBase<Utf8> {
+    static from(values: string[]) {
```

Comment:
We have a bunch of code internally at Graphistry that we could contribute that does essentially this for a number of the types. For example, going from a random JS Array to a Dictionary encoded binary or UTF-8 vector, with null handling. We wrote it mostly to convert between existing protobuf data + random JS arrays, so it allows for customizing the encoding a bit with selectors (e.g. you can pass in a function to indicate the string `"\0"` should be treated as a null). Ex: https://gist.github.com/trxcllnt/2b4216fc95288c2e0ff92ecf826376e0#file-arrow-encode-ts-L34

Reply:
@TheNeuralBit I can push that stuff up to this PR if you like

Reply:
oh yeah that would be awesome!